File: Swift/lambda - closure/2/main.swift
//
//  main.swift
//  s
//
//  Created by Fay on 2017/5/26.
//  Copyright © 2017年 Fay. All rights reserved.
//

import Foundation

func say(_ reply: (_ content: String) -> String) {
    print(reply("Hello World!"))
}

say { (content) in
    print(content)
    return "Hello Swift!"
}

File: Scala/lambda - closure/2/main.scala
/**
  * Created by fay on 2017/6/2.
  */

object main {
  def main(args: Array[String]): Unit = {
    say(content => {
      println(content)
      return "Hello Scala!"
    })
  }

  def say(reply: (String) => String): Unit = {
    println(reply("Hello World!"))
  }
}
